chapter_01/p03_urlify.py

```diff
@@ -5,6 +5,11 @@
 def urlify_algo(string, length):
     """replace spaces with %20 and removes trailing spaces"""
     # convert to list because Python strings are immutable
+    
+    # there must be at least two trailing spaces for this algorithm to work
+    if ord(string[-1]) != ord(string[-2]) != 32:
```

Comment:
This doesn't work since a string may have no spaces.  If we did want to enforce this though probably we'd do something like:
```python
if not string.endswith("  "):
    raise ValueError("String needs to end with spaces")
```